app/controllers/sessions/magic_links_controller.rb

```diff
@@ -22,11 +21,41 @@ def code
       params.expect(:code)
     end
 
+    def respond_to_valid_code_from(magic_link)
+      respond_to do |format|
+        format.html do
+          start_new_session_for magic_link.identity
+          redirect_to after_sign_in_url(magic_link)
+        end
+
+        format.json do
+          new_access_token = magic_link.identity.access_tokens.create!(permission: :write)
```

Comment:
Creating a token might not be what we want in this case because they'll show up in the access tokens section.
Maybe we could use the session cookie to authenticate the mobile apps? IDK, how we do auth on other apps, will check that. But the cookie would be the easiest solution.